authorization/invitation/model/invitation_model_service.go

```diff
@@ -0,0 +1,201 @@
+package model
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/fabric8-services/fabric8-auth/account"
+	"github.com/fabric8-services/fabric8-auth/authorization"
+	"github.com/fabric8-services/fabric8-auth/authorization/invitation"
+	invRepo "github.com/fabric8-services/fabric8-auth/authorization/invitation/repository"
+	permissionModel "github.com/fabric8-services/fabric8-auth/authorization/permission/model"
+	"github.com/fabric8-services/fabric8-auth/authorization/repository"
+	resourceRepo "github.com/fabric8-services/fabric8-auth/authorization/resource/repository"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/satori/go.uuid"
+)
+
+type InvitationModelService interface {
+	Issue(ctx context.Context, repo repository.Repositories, issuingUserId uuid.UUID, inviteTo string, invitations []invitation.Invitation) error
+}
+
+// GormInvitationModelService is the implementation of the interface for
+// InvitationService. IMPORTANT NOTE: Transaction control is not provided by this service
+type GormInvitationModelService struct {
+	permModelSvc permissionModel.PermissionModelService
+}
+
+// NewInvitationModelService creates a new service.
+func NewInvitationModelService(permissionSvc permissionModel.PermissionModelService) InvitationModelService {
+	return &GormInvitationModelService{
+		permModelSvc: permissionSvc,
+	}
+}
+
+// Issue creates new invitations.  The inviteTo parameter is the unique id of the organization, team, security group or resource for
+// which the invitations will be issued, and the invitations parameter contains the users and state for each individual user invitation.
+// This method creates one record in the INVITATION table for each user in the invitations parameter.  Any roles that are issued
+// as part of a user's invitation are created in the INVITATION_ROLE table.
+func (s *GormInvitationModelService) Issue(ctx context.Context, repo repository.Repositories, issuingUserId uuid.UUID, inviteTo string,
+	invitations []invitation.Invitation) error {
+
+	var inviteToIdentity *account.Identity
+	var identityResource *resourceRepo.Resource
+	var inviteToResource *resourceRepo.Resource
+
+	// First try to convert inviteTo to a uuid
+	inviteToUUID, err := uuid.FromString(inviteTo)
+	// If we get an error here, the value is definitely not for an Identity so we'll treat it as a resource ID
+	if err != nil {
+		// Try to lookup a resource with the same ID value
+		inviteToResource, err = repo.ResourceRepository().Load(ctx, inviteTo)
```

Comment:
Well, since the Resource ID is the primary key for the RESOURCE table, we already have a unique constraint the same as every other table's primary key.  Meaning that attempting to create a resource with an ID that already exists will fail with an error.